widgets/AttributesTable/_QueryTaskMixin.js

```diff
@@ -437,6 +437,18 @@ define([
                             url = layer.url + '/' + layer.visibleLayers[0];
                         }
                     }
+                    
+                    if (qp.includeLayerDefinitions && layer.layerDefinitions) {
```

Comment:
Feature layers do not have a `layerDefinitions` property (at least not documented). I believe the [`getDefinitionExpression`](https://developers.arcgis.com/javascript/3/jsapi/featurelayer-amd.html#getdefinitionexpression) method is the right one to use for feature layers. Dynamic layers do not have a comparable `get` method so you have to use this property.